client/src/document/toolbar/flaws.tsx

```diff
@@ -4,14 +4,101 @@ import { annotate, annotationGroup } from "rough-notation";
 import { RoughAnnotation } from "rough-notation/lib/model";
 
 import { humanizeFlawName } from "../../flaw-utils";
-import { Doc, Link, MacroErrorMessage, BadBCDLink } from "../types";
+import {
+  Doc,
+  Link,
+  MacroErrorMessage,
+  BadBCDLink,
+  ImageReferenceFlaw,
+  GenericFlaw,
+} from "../types";
 import "./flaws.scss";
 
 interface FlawCount {
   name: string;
   count: number;
 }
 
+function useAnnotations(genericFlaws: GenericFlaw[]) {
+  useEffect(() => {
+    const annotations: RoughAnnotation[] = [];
+    const elements: HTMLElement[] = [];
+    for (const flaw of genericFlaws) {
+      const element: HTMLElement | null = document.querySelector(
```

Comment:
nit: unnecesary type annotation, that is the exact return type of querySelector

Reply:
Not entirely true. `document.querySelector` returns `Element | null` which means things like `element.title` won't work. 

But, on second thought I think the `as` notation feels more appropriate...
```typescript
      const element = document.querySelector(
        `[data-flaw="${flaw.id}"]`
      ) as HTMLElement;
```
So I'll switch to that for now.